storage/engine.go

```diff
@@ -155,7 +157,7 @@ func (e *Engine) WithLogger(log *zap.Logger) {
 // PrometheusCollectors returns all the prometheus collectors associated with
 // the engine and its components.
 func (e *Engine) PrometheusCollectors() []prometheus.Collector {
```

Comment:
This method is already being called [here](https://github.com/influxdata/influxdb/blob/master/cmd/influxd/launcher/launcher.go#L854) in the `influxd` launcher, so all the collectors we return should get registered in the prometheus HTTP registry